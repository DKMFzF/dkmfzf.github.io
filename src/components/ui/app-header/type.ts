import { TLinks } from '../icon/type';

export type TAppHeaderUIProps = {
	links: TLinks;
};
