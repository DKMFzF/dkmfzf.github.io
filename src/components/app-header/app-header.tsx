import { FC } from 'react';
import { AppHeaderUI } from '../ui';
import { links } from '@utils-constants';

export const AppHeader: FC = () => <AppHeaderUI links={links} />;
