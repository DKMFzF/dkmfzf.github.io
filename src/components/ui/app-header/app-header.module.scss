@use '~styles/variables' as *;

.header {
	position: fixed;
	inset: 10px 0 0;
	z-index: 1;
	inline-size: 100%;

	&__menu {
		display: flex;
		flex-direction: row;
		justify-content: space-between;
		font-family: $font;
	}
}
