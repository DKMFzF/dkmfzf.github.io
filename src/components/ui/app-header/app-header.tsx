import { FC } from 'react';
import styles from './app-header.module.scss';
import { TAppHeaderUIProps } from './type';
import { ProfileIcon } from '@zlden/react-developer-burger-ui-components';
import { Link, useLocation } from 'react-router-dom';
import logoImage from '../../../image/logo_line.svg';
import React from 'react';

/**
 * Упрощенный компонент шапки приложения
 * Содержит только две ссылки по краям:
 * - Слева: Оформление заказа
 * - Справа: Личный кабинет
 */
export const AppHeaderUI: FC<TAppHeaderUIProps> = ({ userName }) => {
	const location = useLocation();

	const isActiveLink = (path: string, exact = false) =>
		exact ? location.pathname === path : location.pathname.startsWith(path);

	return <header />;
};
