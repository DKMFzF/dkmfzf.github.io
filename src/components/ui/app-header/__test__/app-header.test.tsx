import { expect, it, jest } from '@jest/globals';
import { AppHeaderUI } from '../app-header';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

jest.mock(
	'../../../../../public/kirill-doroshev-resume.pdf',
	() => 'test-resume.pdf'
);

describe('[AppHeaderUI]: test render', () => {
	it('Renders header correctly and matches snapshot', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<AppHeaderUI
					links={{
						github: '',
						email: '',
						telegram: ''
					}}
				/>
			</MemoryRouter>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
