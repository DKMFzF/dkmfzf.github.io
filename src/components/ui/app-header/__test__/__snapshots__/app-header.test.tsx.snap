// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AppHeaderUI]: test render Renders header correctly and matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="header__container"
    >
      <div
        class="header__inside-wrapper"
      >
        <div
          class="header__brand"
        >
          <a
            class="header__logo"
            data-cy="link-portfolio"
            data-discover="true"
            data-text="portfolio"
            href="/"
          >
            portfolio
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__menu"
          >
            <li
              class="header__menuItem"
            >
              <a
                class="header__link"
                data-discover="true"
                href="/"
              >
                About Me
              </a>
            </li>
            <li
              class="header__menu-item"
            >
              <a
                class="header__link"
                data-discover="true"
                href="/portfolio"
              >
                portfolio
              </a>
            </li>
            <li
              class="header__menu-item"
            >
              <a
                class="header__link"
                download="kirill-doroshev-resume.pdf"
                href="test-resume.pdf"
              >
                download resume
              </a>
            </li>
            <li
              class="header__menu-item"
            >
              <div
                class="header__icons"
              >
                <div
                  style="display: flex; gap: 16px;"
                />
              </div>
            </li>
          </ul>
        </nav>
        <div
          class="header__mobile-menu"
        >
          <button
            class="header__burger-button"
          >
            â˜°
          </button>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
